plugins/kubernetes/app/models/watchers/base_cluster_watcher.rb

```diff
@@ -8,8 +8,9 @@ class BaseClusterWatcher
 
     finalizer :stop_watching
 
-    def initialize(watch_stream)
-      @watch_stream = watch_stream
+    def initialize(args)
+      @cluster = args[:cluster]
+      @watch_stream = args[:watch_stream]
```

Comment:
Now I'm the one confused. :)

What do you have in mind with composition in this case? Making `ClusterPodWatcher` and `ClusterErrorPodWatcher` own a ClusterWatcher? I can't see what else could be extracted.
I know Composition tends to be better than inheritance, but in this case... I'm missing how it could make this more clean.

An alternative to creating the hash would be to pass only the cluster and have a template method on the BaseClusterWatcher, overrided in the child classes, that given a cluster returns the required stream. Would you prefer this approach?